### DISCUSSION TITLE
Json Export feature

### DISCUSSION THREAD

**Discussion by asbiin** [monicahq/monica#5824] [General]
**Json export is in preview mode. Tell us what you think about it.**

This feature allows you to export all your data as a json file.
There is no `import` feature for now.

Found export in Settings > Export data of your application, like [https://app.monicahq.com/settings/export](https://app.monicahq.com/settings/export).

**Comment by Neriderc** [comment]
Looking good! One of my concerns with Monica has been the "exit plan". What happens in 10 years (or 20) when I've accumulated all this data, but the software isn't being maintained and there aren't any viable forks? I have all this valuable data, but eventually I won't be able to access it anymore.

Having this export means I can at least be confident that I can take the data in a standard format and pull what I need from it, in a way that the SQL export _may_ have allowed but I'm happier to see this JSON option appear.

One suggestion for user friendliness, I click the button to export and after 30 seconds I received a timeout error. I thought it hadn't worked, until I made my way back to the page and found it was running in the background, and eventually I could see a download link. I'd suggest maybe just triggering the background process and maybe show a message saying it has started, maybe add the process to the page without a refresh, something like that.

If it helps, I'm running the official docker image on a Raspberry Pi 4 (8GB).

**Reply by Brice187** [reply]
Import Option would be nice

**Reply by asbiin** [reply]
> One suggestion for user friendliness, I click the button to export and after 30 seconds I received a timeout error. I thought it hadn't worked, until I made my way back to the page and found it was running in the background, and eventually I could see a download link. I'd suggest maybe just triggering the background process and maybe show a message saying it has started, maybe add the process to the page without a refresh, something like that.

SQL export can take some times. It's working well on a batched job (with a queue). When you have a queue, there is no timeout on the UI, the job will indeed run in the background, and the user will get a notification. But I understand that it would be nice to have a "refresh" button, or if the table can be refreshed automatically.

Thank you

**Comment by cstrat** [comment]
I am always a fan of being able to export data and like @Neriderc said, JSON is a great addition to the SQL export.

My instance also timed out after 30 seconds and I thought it wasn't working.
I am still waiting but it shows this when I navigate back.

![image](https://user-images.githubusercontent.com/4209076/152253833-266baea9-17c7-48f6-9629-2ca1b8ad0680.png)

Similar to @Neriderc I am running my own instance (supporting via Patreon).

**Reply by asbiin** [reply]
Json export seems to be memory-consuming. It's working well on batched job (with a queue). Definitely a good way to improve performances of your instance!

**Reply by cstrat** [reply]
Thanks for the continued development. Mine works, but I appear to have a stuck attempt.

![image](https://user-images.githubusercontent.com/4209076/155066196-e3241379-3e5e-4df1-bcd2-995902ae179d.png)

Surely this doesn't matter, but it would be nice if I could clear this.

**Comment by jatls** [comment]
JSON Export worked a treat on a self-hosted instance. Even got an email notification when complete. Nice touch!

Is the roadmap to replace the SQL export with JSON? Or to have both available in to the future?

**Reply by asbiin** [reply]
I don't think we will maintain SQL export when JSON export is working 100%. SQL export current implementation is not reliable, it's limited to the exported data, it does not contain extra-resources like photos and documents, ...

**Comment by AndyYangUK** [comment]
Just tried it and it worked well. Can't wait for the import function when I inevitably mess up the install and have to start again!

**Comment by aluhrs13** [comment]
I really like this. I have scripts that make JSON from as much of my personal data as possible so I can transform them into different formats and search across sources. Import of course would be nice, but export is definitely appreciated.

**Reply by crazyboy24** [reply]
can you share that script?

**Comment by osvald-laurits** [comment]
Love it! Opening and searching json is much more comfortable than sql.
To download the json I click on "Export to Json" then open my email inbox and wait for the email notification and then click the link in the email notification. At the first time using it I thought that the "Submitted" Status would automatically change when the download is ready. Refreshing the page is needed for the Status to update.

**Comment by ChargingBulle** [comment]
Both exports are not working for me

![image](https://user-images.githubusercontent.com/6533815/168687830-5a3906b9-e8fb-4fc5-bc6e-1630747451bc.png)

**Reply by wolf-rainer** [reply]
Same issue here. 
See workarounfd from https://github.com/monicahq/monica/issues/5917 - even though FE showing failed, the exports are created and available from app container.

**Comment by fruwajek** [comment]
Hello,
on my instance, I hava the same problem, every export reported a bug.
I use Monica v 3.7.0 on Docker in my RPi with mariaDB.

![Monica_export_2](https://user-images.githubusercontent.com/107408546/173358933-76e1acac-77ad-4f5a-bfe2-96add5a6d55a.PNG)

Mariadb doesn't show any log about export data, someone have a solution of this problem?
Thanks.

**Reply by wolf-rainer** [reply]
Not sure about MariaDB, but maybe check the answer above: 
Check if your exports in fact are successfully created and listed in the app container under /var/www/html/storage/app/public/exports/.

**Comment by wolf-rainer** [comment]
Also from me both thumbs up for JSON export!
Great feature!

Although having 'Failed' after ca. 30 seconds issue for both export types in GUI, the files ARE created and accessible in app container. (I'm running Monica 3.7.0 on Raspberry cluster Docker swarm). 

JSON is so much easier to parse and reuse in any programming language.

**Comment by expionage** [comment]
+1 same issue.

I logged this as well as Issue #6441.

**Comment by teslazap** [comment]
I have a fresh official docker install and 1 contact

![image](https://user-images.githubusercontent.com/10651478/221603092-b3d353ac-2f8f-43c1-a191-2583753e729d.png)

**Reply by hemps69** [reply]
I have the same issue, all my exports are failing using either method and using version Version: 4.1.2 hosted on pikapod.net. Any suggestion would be great I have 132 contacts I would like to backup.

**Comment by kyeotic** [comment]
Is import feature planned?
